kernels/deinterleave.cu

```diff
@@ -0,0 +1,65 @@
+#include <complex>
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <cusp/deinterleave.cuh>
+#include <cusp/helper_cuda.h>
+
+namespace cusp {
+
+// Might want to expand uint8_t interface to maximize bus throughput
+__global__ void kernel_deinterleave(const uint8_t *in, uint8_t **outs,
+                                    int nstreams, int itemsperblock,
+                                    int itemsize, int N) {
+  int i = blockIdx.x * blockDim.x + threadIdx.x;
+  if (i < N) {
+    int a = itemsize * itemsperblock;
+    int stream_idx = (i / a) % nstreams;
```

Comment:
Integer modulo is extremely slow on GPU's, so it might make sense to define our own modulo operator.

See: https://stackoverflow.com/questions/33333363/built-in-mod-vs-custom-mod-function-improve-the-performance-of-modulus-op